### PR TITLE
core, internal, eth: add blob tx to eth_call and eth_estimategas rpc methods

### PR DESCRIPTION
**Pull request by andicrypt**

### CODE REVIEW

File: core/types/transaction_test.go [hunk 1]
```diff
@@ -268,10 +271,20 @@ func TestTransactionCoding(t *testing.T) {
...
+		eip2930Signer = NewEIP2930Signer(common.Big1)
+		// londonSigner  = NewLondonSigner(common.Big1)
+		cancunSigner = NewCancunSigner(common.Big1)
 	)
 	for i := uint64(0); i < 500; i++ {
 		var txdata TxData
-		switch i % 6 {
+		switch i % 7 {
+		case 6:
+			signer = cancunSigner
```
Comment (minh-bq) [inline core/types/transaction_test.go:283]: Why don't we just change signer to always be cancunSigner?

File: eth/gasestimator/gasestimator.go [hunk 2]
```diff
@@ -80,6 +80,15 @@ func Estimate(ctx context.Context, call core.Message, opts *Options, gasCap uint
 			}
 			available.Sub(available, call.Value())
 		}
+		if opts.Config.IsCancun(opts.Header.Number) {
+			blobGasPerBlob := new(big.Int).SetUint64(params.BlobTxBlobGasPerBlob)
+			blobBalanceUsage := new(big.Int).SetUint64(uint64(len(call.BlobHashes())))
+			blobBalanceUsage.Mul(blobBalanceUsage, blobGasPerBlob)
+			blobBalanceUsage.ModInverse(blobBalanceUsage, call.BlobGasFeeCap())
```
Comment (minh-bq) [inline eth/gasestimator/gasestimator.go:87]: Why do we use ModInverse here?

File: internal/ethapi/api.go [hunk 3]
```diff
@@ -468,6 +470,9 @@ func (s *PrivateAccountAPI) SendTransaction(ctx context.Context, args Transactio
 		s.nonceLock.LockAddr(args.from())
 		defer s.nonceLock.UnlockAddr(args.from())
 	}
+	if args.IsEIP4844() {
```
Comment (minh-bq) [inline internal/ethapi/api.go:473]: Why don't we support EIP 4844 here?

File: eth/gasestimator/gasestimator.go (hunk 2 above)
Comment (andicrypt) [reply eth/gasestimator/gasestimator.go:87]: Fix, it's mistaken coding.

File: internal/ethapi/transaction_args.go [hunk 4]
```diff
@@ -239,7 +294,7 @@ func (args *TransactionArgs) ToMessage(globalGasCap uint64, baseFee *big.Int) (t
 	if args.AccessList != nil {
 		accessList = *args.AccessList
 	}
-	msg := types.NewMessage(addr, args.To, 0, value, gas, gasPrice, gasFeeCap, gasTipCap, data, accessList, true)
+	msg := types.NewMessage(addr, args.To, 0, value, gas, gasPrice, gasFeeCap, gasTipCap, data, accessList, true, (*big.Int)(args.BlobFeeCap), args.BlobHashes)
```
Comment (minh-bq) [inline internal/ethapi/transaction_args.go:297]: What if user inputs args.BlobHashes != nil but args.BlobFeeCap == nil, does it lead to a panic?

Comment (andicrypt) [reply internal/ethapi/transaction_args.go:297]: Can you give more context for this

**Review by minh-bq — APPROVED** [review]

**Pull request merged**